mqtt/mqtt-bridge/src/controller.rs

```diff
@@ -1,47 +1,48 @@
-use std::collections::HashMap;
-
 use futures_util::future::{self, join_all};
+use tokio::task;
 use tracing::{error, info};
 
-use crate::bridge::{Bridge, BridgeError};
-use crate::settings::Settings;
+use crate::bridge::Bridge;
+use crate::settings::BridgeSettings;
 
 /// Controller that handles the settings and monitors changes, spawns new Bridges and monitors shutdown signal.
-#[derive(Default)]
-pub struct BridgeController {
-    bridges: HashMap<String, Bridge>,
-}
+pub struct BridgeController {}
 
 impl BridgeController {
-    pub fn new() -> Self {
-        Self::default()
-    }
+    pub async fn run(system_address: String, device_id: String, settings: BridgeSettings) {
+        info!("starting bridge controller...");
 
-    pub async fn init(
-        &mut self,
-        system_address: String,
-        device_id: &str,
-    ) -> Result<(), BridgeError> {
-        info!("initializing bridge controller...");
-        let settings = Settings::new().map_err(BridgeError::LoadingSettings)?;
-
-        if let Some(upstream) = settings.upstream() {
-            let bridge = Bridge::new(system_address, device_id.into(), upstream.clone());
-            self.bridges.insert(upstream.name().to_string(), bridge);
+        let mut bridge_handles = vec![];
+        if let Some(upstream_settings) = settings.upstream() {
+            let upstream_settings = upstream_settings.clone();
+            let upstream_bridge = async move {
+                let bridge =
+                    Bridge::new(system_address, device_id.into(), upstream_settings.clone()).await;
+
+                // TODO REVIEW: Better way to log?
+                match bridge {
```

Comment:
There is a better way to log err with format params, but I didn't see it from the docs.